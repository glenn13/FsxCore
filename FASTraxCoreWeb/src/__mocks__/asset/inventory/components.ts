import {AxiosResponse} from 'axios';
import {QueryResult} from 'react-query';
import {Component, newComponent} from '../../../entities/asset/inventory/Component';

export const mockComponentsQuery = (): QueryResult<AxiosResponse<Component[]>> =>
  ({data: {data: [newComponent(), newComponent(), newComponent()]}} as QueryResult<
    AxiosResponse<Component[]>
  >);
