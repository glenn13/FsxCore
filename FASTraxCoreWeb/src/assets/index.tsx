const ImageResources = {
  Fastrax: require('./images/Fastrax.png'),
  FastraxLogo: require('./images/fastrax-logo.svg'),
  FastraxWhite: require('./images/Fastrax-white.png'),
  FsxSquareThumb: require('./images/thumb-fsx-squares.png'),
  PageNotFound: require('./images/404@2x.png'),
  AMSLogo: require('./images/AMS-logo.png'),
  Tracking: require('./images/tracking.svg'),
  MaintenanceSpareParts: require('./images/maintenance-spare-parts.svg'),
  NoNotificationFound: require('./images/no-notification-found.svg'),
  Verification: require('./images/verifycode.png'),
};

export {ImageResources};
