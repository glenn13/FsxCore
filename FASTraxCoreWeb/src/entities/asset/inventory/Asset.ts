import * as yup from 'yup';

import {Project} from '../../catalog/Project';
import {generateUUID} from './../../../helpers/randoms';
import {EntityCustomField} from '../../global/EntityCustomField';
import {MaintenanceStatus} from '../standard-entries/MaintenanceStatus';
import {AssetOwnership, assetOwnershipShape, newAssetOwnership} from './AssetOwnership';

export const newAsset = (): Asset => ({
    id: 0,
    assetRefId: generateUUID(true).toUpperCase(),
    projectSiteId: 0,
    assetManufacturerId: 0,
    assetStatusId: 0,
    assetGroupId: 0,
    assetModelId: 0,
    maintenanceStatusId: 0,
    assetTypeId: 0,
    seriesId: 0,
    assetOwnershipTypeId: 0,
    assetModelYearId: 0,
    assetColorId: 0,
    assetOwnershipId: 0,
    lastServiceDate: new Date(),
    inventoryDate: new Date(),
    dateRegistered: new Date(),
    serialNo: '',
    title: '.',
    assetOwnership: newAssetOwnership(),
});

export interface Asset {
  id: number;
  projectSiteId: number;
  assetRefId: string;
  assetManufacturerId: number;
  assetStatusId: number;
  assetGroupId: number;
  assetTypeId: number;
  assetModelId: number;
  maintenanceStatusId: number;
  seriesId: number;
  assetOwnershipTypeId: number;
  assetModelYearId: number;
  assetColorId: number;
  lastServiceDate: Date;
  inventoryDate: Date;
  dateRegistered: Date;
  serialNo: string;
  title: string;
  description?: string;
  assetOwnershipId: number;

  base64stringImage?: string;
  uri?: string;

  projectSite?: Project;
  assetCategory?: AssetCategory;
  assetManufacturer?: AssetManufacturer;
  assetStatus?: AssetStatus;
  assetGroup?: AssetGroup;
  assetModel?: AssetModel;
  maintenanceStatus?: MaintenanceStatus;
  assetType?: AssetType;
  series?: AssetSeries;
  assetOwnershipType?: AssetOwnershipType;
  assetModelYear?: AssetModelYear;
  assetColor?: AssetColor;
  assetOwnership: AssetOwnership;

  assetCustomFields?: EntityCustomField[];
}

export const assetShape = {
  assetRefId: yup.string().required(),
  assetManufacturerId: yup.number().min(1, 'Manufacturer is required!').required(),
  assetStatusId: yup.number().min(1, 'Asset Status is required!').required(),
  assetGroupId: yup.number().min(1, 'Asset Group is required!').required(),
  assetTypeId: yup.number().min(1, 'Asset Type is required!').required(),
  assetModelId: yup.number().min(1, 'Model is required!').required(),
  maintenanceStatusId: yup.number().min(1, 'Maintenance Status is required!').required(),
  seriesId: yup.number().min(1, 'Series is required!').required(),
  assetOwnershipTypeId: yup.number().min(1, 'Ownership Type is required!').required(),
  assetModelYearId: yup.number().min(1, 'Model Year is required!').required(),
  assetColorId: yup.number().min(1, 'Color is required!').required(),
  lastServiceDate: yup.date(),
  inventoryDate: yup.date().required(),
  dateRegistered: yup.date().required(),
  serialNo: yup.string().required('Serial No. is required!'),
  title: yup.string().required(),
  description: yup.string().nullable(),
  assetOwnership: yup.object().shape(assetOwnershipShape),
};
