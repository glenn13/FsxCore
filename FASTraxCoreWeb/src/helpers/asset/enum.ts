export enum AssetCategoryEnum {
    GeneralAsset = 1,
    Vehicle = 2,
    Component = 3    
}

export enum TransactionHistoryTypeEnum {
    Disposition = 1,
    Estimate = 2,
    Inspection = 3,    
    WorkOrder = 4,
}

export enum DispositionStatusEnum {
    Open = 1,
    InProgress = 2,
    Approved = 3,
    Rejected = 4
}
