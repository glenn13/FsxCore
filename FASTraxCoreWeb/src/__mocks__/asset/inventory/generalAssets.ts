import {AxiosResponse} from 'axios';
import {QueryResult} from 'react-query';
import {createAxiosResponse} from './utils';
import {newGeneralAssetGroup} from '@app/entities/asset/inventory/GeneralAssetGroup';
import {GeneralAsset, newGeneralAsset} from '../../../entities/asset/inventory/GeneralAsset';

export const mockGeneralAssetsQuery = (): QueryResult<AxiosResponse<GeneralAsset[]>> =>
  ({data: {data: [newGeneralAsset()]}} as QueryResult<AxiosResponse<GeneralAsset[]>>);

export const mockGeneralAssetsGroups = () =>
  createAxiosResponse([newGeneralAssetGroup(), newGeneralAssetGroup(), newGeneralAssetGroup()]);

export const mockGeneralAssets = (): GeneralAsset[] => [newGeneralAsset(), newGeneralAsset()];
