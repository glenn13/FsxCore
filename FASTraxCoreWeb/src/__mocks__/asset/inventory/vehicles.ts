import {AxiosResponse} from 'axios';
import {QueryResult} from 'react-query';
import {Vehicle, newVehicle} from '../../../entities/asset/inventory/Vehicle';

export const mockVehiclesQuery = (): QueryResult<AxiosResponse<Vehicle[]>> =>
  ({data: {data: [newVehicle(), newVehicle()]}} as QueryResult<AxiosResponse<Vehicle[]>>);
