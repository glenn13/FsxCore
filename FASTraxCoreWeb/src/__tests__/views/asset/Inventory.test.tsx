import React from 'react';
import thunk from 'redux-thunk';
import {Formik, Form} from 'formik';
import {Provider} from 'react-redux';
import {render, waitFor} from '@testing-library/react';
import configureStore, {MockStoreEnhanced} from 'redux-mock-store';
import GeneralAssetsTab from '@app/views/asset/Inventory/GeneralAssets';
import VehiclesTab from '@app/views/asset/Inventory/Vehicles';
import ComponentsTab from '@app/views/asset/Inventory/Components';
import {mockGeneralAssetsGroups} from '@app/__mocks__/asset/inventory/generalAssets';
import {mockVehiclesQuery} from '@app/__mocks__/asset/inventory/vehicles';
import {mockComponentsQuery} from '@app/__mocks__/asset/inventory/components';
import {initialMockState, RootState} from '@app/store/rootReducer';
import * as generalAssetService from '@app/services/asset/assets.service';
import * as vehicleService from '@app/services/asset/vehicles.service';
import * as componentService from '@app/services/asset/inventory/components.service';
import AssetForm from '@app/views/asset/Inventory/AssetForm';
import {newAsset} from '@app/entities/asset/inventory/Asset';
import {renderWithTheme} from '@app/helpers/test';
import { AssetCategoryEnum } from '@app/helpers/asset/enum'

let store: MockStoreEnhanced;
const middlewares = [thunk];
const mockStore = configureStore<RootState>(middlewares);

beforeEach(() => {
  store = mockStore(initialMockState);
});

describe('GeneralAssetsTab Component', () => {
  it('renders table of General Asset Groups', async () => {
    jest
      .spyOn(generalAssetService, 'getGeneralAssetGroups')
      .mockResolvedValueOnce(mockGeneralAssetsGroups());

    const {getByTestId, asFragment} = render(
      <Provider store={store}>
        <GeneralAssetsTab />
      </Provider>,
    );

    const node = await waitFor(() => getByTestId('generalAssetGroups'));
    expect(node.querySelectorAll('.k-master-row').length).toBe(3);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('VehiclesTab Component', () => {
  it('renders table of Vehicles', async () => {
    jest.spyOn(vehicleService, 'useVehiclesGrid').mockImplementation(mockVehiclesQuery);
    const wrapper = render(
      <Provider store={store}>
        <VehiclesTab />
      </Provider>,
    );

    expect(wrapper.container.querySelectorAll('.k-master-row').length).toBe(2);
  });
});

describe('ComponentsTab Component', () => {
  it('renders table of Components', async () => {
    jest.spyOn(componentService, 'useComponentsGrid').mockImplementation(mockComponentsQuery);
    const wrapper = render(
      <Provider store={store}>
        <ComponentsTab />
      </Provider>,
    );

    expect(wrapper.container.querySelectorAll('.k-master-row').length).toBe(3);
  });
});

describe('AssetForm Component', () => {
  const initialValues = newAsset();
  it('displays VIN as label when category is Vehicle', () => {
    const {getByText, queryByText} = renderWithTheme(
      <Formik initialValues={initialValues} onSubmit={() => {}}>
            <Form>
                <AssetForm images={[]} categoryId={AssetCategoryEnum.Vehicle} onCategoryChange={() => { }} />
        </Form>
      </Formik>,
    );

    expect(getByText(new RegExp('VIN', 'i'))).toBeInTheDocument();
    expect(queryByText(new RegExp('Serial No', 'i'))).not.toBeInTheDocument();
  });

  it('displays Serial No as label when category is General Asset', () => {
    const {getByText, queryByText} = renderWithTheme(
      <Formik initialValues={initialValues} onSubmit={() => {}}>
            <Form>
                <AssetForm images={[]} categoryId={AssetCategoryEnum.GeneralAsset} onCategoryChange={() => { }} />
        </Form>
      </Formik>,
    );

    expect(queryByText(new RegExp('VIN', 'i'))).not.toBeInTheDocument();
    expect(getByText(new RegExp('Serial No', 'i'))).toBeInTheDocument();
  });

  it('displays Serial No as label when category is Components', () => {
    const {getByText, queryByText} = renderWithTheme(
      <Formik initialValues={initialValues} onSubmit={() => {}}>
            <Form>
                <AssetForm images={[]} categoryId={AssetCategoryEnum.GeneralAsset} onCategoryChange={() => { }} />
        </Form>
      </Formik>,
    );

    expect(queryByText(new RegExp('VIN', 'i'))).not.toBeInTheDocument();
    expect(getByText(new RegExp('Serial No', 'i'))).toBeInTheDocument();
  });
});
