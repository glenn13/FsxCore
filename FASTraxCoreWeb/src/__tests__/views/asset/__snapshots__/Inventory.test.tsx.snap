// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneralAssetsTab Component renders table of General Asset Groups 1`] = `
<DocumentFragment>
  <div
    class="pt-2"
  >
    <div
      class="flex flex-1 flex-col"
      data-testid="generalAssetGroups"
    >
      <div
        class="k-widget k-grid"
        role="grid"
      >
        <div
          class="k-grid-header"
          role="presentation"
          style="padding: 0px 0px 0px 0px;"
        >
          <div
            class="k-grid-header-wrap"
            role="presentation"
            style="border-width: 0px;"
          >
            <table
              role="presentation"
            >
              <colgroup
                role="presentation"
              >
                <col />
                <col />
                <col />
                <col />
                <col />
                <col />
              </colgroup>
              <thead
                role="presentation"
              >
                <tr
                  role="row"
                >
                  <th
                    aria-colindex="1"
                    aria-selected="false"
                    aria-sort="none"
                    class="k-header "
                    colspan="1"
                    role="columnheader"
                    rowspan="1"
                  >
                    <a
                      class="k-link"
                    >
                      Group No
                    </a>
                  </th>
                  <th
                    aria-colindex="2"
                    aria-selected="false"
                    aria-sort="none"
                    class="k-header "
                    colspan="1"
                    role="columnheader"
                    rowspan="1"
                  >
                    <a
                      class="k-link"
                    >
                      General Asset Name
                    </a>
                  </th>
                  <th
                    aria-colindex="3"
                    aria-selected="false"
                    aria-sort="none"
                    class="k-header "
                    colspan="1"
                    role="columnheader"
                    rowspan="1"
                  >
                    <a
                      class="k-link"
                    >
                      Available Qty
                    </a>
                  </th>
                  <th
                    aria-colindex="4"
                    aria-selected="false"
                    aria-sort="none"
                    class="k-header "
                    colspan="1"
                    role="columnheader"
                    rowspan="1"
                  >
                    <a
                      class="k-link"
                    >
                      Issued Qty
                    </a>
                  </th>
                  <th
                    aria-colindex="5"
                    aria-selected="false"
                    aria-sort="none"
                    class="k-header "
                    colspan="1"
                    role="columnheader"
                    rowspan="1"
                  >
                    <a
                      class="k-link"
                    >
                      Damaged Qty
                    </a>
                  </th>
                  <th
                    aria-colindex="6"
                    aria-selected="false"
                    aria-sort="none"
                    class="k-header "
                    colspan="1"
                    role="columnheader"
                    rowspan="1"
                  >
                    <a
                      class="k-link"
                    >
                      Total Qty
                    </a>
                  </th>
                </tr>
              </thead>
            </table>
          </div>
        </div>
        <div
          class="k-grid-container"
          role="presentation"
        >
          <div
            class="k-grid-content k-virtual-content"
            role="presentation"
          >
            <div
              role="presentation"
              style="position: relative;"
            >
              <table
                class="k-grid-table"
                role="presentation"
                tabindex="-1"
              >
                <colgroup
                  role="presentation"
                >
                  <col />
                  <col />
                  <col />
                  <col />
                  <col />
                  <col />
                </colgroup>
                <tbody
                  role="presentation"
                >
                  <tr
                    class="k-master-row"
                    role="row"
                  >
                    <td
                      aria-colindex="1"
                      aria-selected="false"
                      class=""
                      colspan="1"
                      role="gridcell"
                    />
                    <td
                      aria-colindex="2"
                      aria-selected="false"
                      class=""
                      colspan="1"
                      role="gridcell"
                    />
                    <td
                      aria-colindex="3"
                      aria-selected="false"
                      class=""
                      colspan="1"
                      role="gridcell"
                    >
                      0
                    </td>
                    <td
                      aria-colindex="4"
                      aria-selected="false"
                      class=""
                      colspan="1"
                      role="gridcell"
                    >
                      0
                    </td>
                    <td
                      aria-colindex="5"
                      aria-selected="false"
                      class=""
                      colspan="1"
                      role="gridcell"
                    >
                      0
                    </td>
                    <td
                      aria-colindex="6"
                      aria-selected="false"
                      class=""
                      colspan="1"
                      role="gridcell"
                    />
                  </tr>
                  <tr
                    class="k-master-row k-alt"
                    role="row"
                  >
                    <td
                      aria-colindex="1"
                      aria-selected="false"
                      class=""
                      colspan="1"
                      role="gridcell"
                    />
                    <td
                      aria-colindex="2"
                      aria-selected="false"
                      class=""
                      colspan="1"
                      role="gridcell"
                    />
                    <td
                      aria-colindex="3"
                      aria-selected="false"
                      class=""
                      colspan="1"
                      role="gridcell"
                    >
                      0
                    </td>
                    <td
                      aria-colindex="4"
                      aria-selected="false"
                      class=""
                      colspan="1"
                      role="gridcell"
                    >
                      0
                    </td>
                    <td
                      aria-colindex="5"
                      aria-selected="false"
                      class=""
                      colspan="1"
                      role="gridcell"
                    >
                      0
                    </td>
                    <td
                      aria-colindex="6"
                      aria-selected="false"
                      class=""
                      colspan="1"
                      role="gridcell"
                    />
                  </tr>
                  <tr
                    class="k-master-row"
                    role="row"
                  >
                    <td
                      aria-colindex="1"
                      aria-selected="false"
                      class=""
                      colspan="1"
                      role="gridcell"
                    />
                    <td
                      aria-colindex="2"
                      aria-selected="false"
                      class=""
                      colspan="1"
                      role="gridcell"
                    />
                    <td
                      aria-colindex="3"
                      aria-selected="false"
                      class=""
                      colspan="1"
                      role="gridcell"
                    >
                      0
                    </td>
                    <td
                      aria-colindex="4"
                      aria-selected="false"
                      class=""
                      colspan="1"
                      role="gridcell"
                    >
                      0
                    </td>
                    <td
                      aria-colindex="5"
                      aria-selected="false"
                      class=""
                      colspan="1"
                      role="gridcell"
                    >
                      0
                    </td>
                    <td
                      aria-colindex="6"
                      aria-selected="false"
                      class=""
                      colspan="1"
                      role="gridcell"
                    />
                  </tr>
                </tbody>
              </table>
            </div>
            <div
              class="k-height-container"
              role="presentation"
            >
              <div />
            </div>
          </div>
        </div>
        <div
          class="k-pager-wrap k-pager k-widget k-grid-pager k-pager-sm"
          role="navigation"
        >
          <a
            class="k-link k-pager-nav k-pager-first k-state-disabled"
            href="#"
            title="Go to the first page"
          >
            <span
              aria-label="Go to the first page"
              class="k-icon k-i-arrow-end-left"
            />
          </a>
          <a
            class="k-link k-pager-nav k-state-disabled"
            href="#"
            title="Go to the previous page"
          >
            <span
              aria-label="Go to the previous page"
              class="k-icon k-i-arrow-60-left"
            />
          </a>
          <div
            class="k-pager-numbers-wrap"
          >
            <select
              class="k-dropdown"
            >
              <option
                class="k-link"
              >
                1
              </option>
            </select>
            <ul
              class="k-pager-numbers k-reset"
            >
              <li>
                <a
                  class="k-link k-state-selected"
                  href="#"
                >
                  1
                </a>
              </li>
            </ul>
          </div>
          <a
            class="k-link k-pager-nav k-state-disabled"
            href="#"
            title="Go to the next page"
          >
            <span
              aria-label="Go to the next page"
              class="k-icon k-i-arrow-60-right"
            />
          </a>
          <a
            class="k-link k-pager-nav k-pager-last k-state-disabled"
            href="#"
            title="Go to the last page"
          >
            <span
              aria-label="Go to the last page"
              class="k-icon k-i-arrow-end-right"
            />
          </a>
          <div
            class="k-pager-info k-label"
          >
            1 - 3 of 3 items
          </div>
        </div>
      </div>
    </div>
    <label
      class="text-base tracking-wider uppercase h-12 w-full flex items-center px-4 py-2"
      style="background-color: rgb(242, 242, 242);"
    >
      Secondary Details
    </label>
    <div
      class="flex flex-1 flex-col"
    >
      <div
        class="k-widget k-grid"
        role="grid"
        style="height: 300px;"
      >
        <div
          class="k-grid-header"
          role="presentation"
          style="padding: 0px 0px 0px 0px;"
        >
          <div
            class="k-grid-header-wrap"
            role="presentation"
            style="border-width: 0px;"
          >
            <table
              role="presentation"
            >
              <colgroup
                role="presentation"
              >
                <col />
                <col />
                <col />
                <col />
                <col />
                <col />
                <col />
              </colgroup>
              <thead
                role="presentation"
              >
                <tr
                  role="row"
                >
                  <th
                    aria-colindex="1"
                    aria-selected="false"
                    aria-sort="none"
                    class="k-header "
                    colspan="1"
                    role="columnheader"
                    rowspan="1"
                  >
                    <a
                      class="k-link"
                    >
                      Serial No.
                    </a>
                  </th>
                  <th
                    aria-colindex="2"
                    aria-selected="false"
                    aria-sort="none"
                    class="k-header "
                    colspan="1"
                    role="columnheader"
                    rowspan="1"
                  >
                    <a
                      class="k-link"
                    >
                      Asset Status
                    </a>
                  </th>
                  <th
                    aria-colindex="3"
                    aria-selected="false"
                    aria-sort="none"
                    class="k-header "
                    colspan="1"
                    role="columnheader"
                    rowspan="1"
                  >
                    <a
                      class="k-link"
                    >
                      Maintenance Status
                    </a>
                  </th>
                  <th
                    aria-colindex="4"
                    aria-selected="false"
                    aria-sort="none"
                    class="k-header "
                    colspan="1"
                    role="columnheader"
                    rowspan="1"
                  >
                    <a
                      class="k-link"
                    >
                      Type
                    </a>
                  </th>
                  <th
                    aria-colindex="5"
                    aria-selected="false"
                    aria-sort="none"
                    class="k-header "
                    colspan="1"
                    role="columnheader"
                    rowspan="1"
                  >
                    <a
                      class="k-link"
                    >
                      Manufacturer
                    </a>
                  </th>
                  <th
                    aria-colindex="6"
                    aria-selected="false"
                    aria-sort="none"
                    class="k-header "
                    colspan="1"
                    role="columnheader"
                    rowspan="1"
                  >
                    <a
                      class="k-link"
                    >
                      Model
                    </a>
                  </th>
                  <th
                    aria-colindex="7"
                    aria-selected="false"
                    aria-sort="none"
                    class="k-header "
                    colspan="1"
                    role="columnheader"
                    rowspan="1"
                  >
                    <a
                      class="k-link"
                    >
                      Model Year
                    </a>
                  </th>
                </tr>
              </thead>
            </table>
          </div>
        </div>
        <div
          class="k-grid-container"
          role="presentation"
        >
          <div
            class="k-grid-content k-virtual-content"
            role="presentation"
          >
            <div
              role="presentation"
              style="position: relative;"
            >
              <table
                class="k-grid-table"
                role="presentation"
                tabindex="-1"
              >
                <colgroup
                  role="presentation"
                >
                  <col />
                  <col />
                  <col />
                  <col />
                  <col />
                  <col />
                  <col />
                </colgroup>
                <tbody
                  role="presentation"
                >
                  <tr
                    class="k-grid-norecords"
                  >
                    <td
                      colspan="7"
                    >
                      No records available
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div
              class="k-height-container"
              role="presentation"
            >
              <div />
            </div>
          </div>
        </div>
        <div
          class="k-pager-wrap k-pager k-widget k-grid-pager k-pager-sm"
          role="navigation"
        >
          <a
            class="k-link k-pager-nav k-pager-first k-state-disabled"
            href="#"
            title="Go to the first page"
          >
            <span
              aria-label="Go to the first page"
              class="k-icon k-i-arrow-end-left"
            />
          </a>
          <a
            class="k-link k-pager-nav k-state-disabled"
            href="#"
            title="Go to the previous page"
          >
            <span
              aria-label="Go to the previous page"
              class="k-icon k-i-arrow-60-left"
            />
          </a>
          <div
            class="k-pager-numbers-wrap"
          >
            <select
              class="k-dropdown"
            />
            <ul
              class="k-pager-numbers k-reset"
            />
          </div>
          <a
            class="k-link k-pager-nav k-state-disabled"
            href="#"
            title="Go to the next page"
          >
            <span
              aria-label="Go to the next page"
              class="k-icon k-i-arrow-60-right"
            />
          </a>
          <a
            class="k-link k-pager-nav k-pager-last k-state-disabled"
            href="#"
            title="Go to the last page"
          >
            <span
              aria-label="Go to the last page"
              class="k-icon k-i-arrow-end-right"
            />
          </a>
          <div
            class="k-pager-info k-label"
          >
            0 - 0 of 0 items
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
